From 761dc462cf66b3307b1b68af49399afed29b1c09 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Fri, 8 Dec 2017 19:32:34 +0800
Subject: [PATCH 3061/5242] MLK-16549-2: ISI: fix suspend/resume error issue

commit  773360b576d7df285fd51712d9a051fe75a2df0d from
https://source.codeaurora.org/external/imx/linux-imx.git

Because ISI runtime suspend has disable clock before system
suspend. If driver does not know that and disable clock again
in system suspend callback. It will lead to isi clk count mis-
match.

Reviewed-by: Sandor.Yu <sandor.yu@nxp.com>
Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-isi-core.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/imx8/mxc-isi-core.c b/drivers/media/platform/imx8/mxc-isi-core.c
index 1ddf3d9..29a1999 100644
--- a/drivers/media/platform/imx8/mxc-isi-core.c
+++ b/drivers/media/platform/imx8/mxc-isi-core.c
@@ -209,7 +209,8 @@ static int mxc_isi_pm_suspend(struct device *dev)
 {
 	struct mxc_isi_dev *mxc_isi = dev_get_drvdata(dev);
 
-	if (mxc_isi->flags & MXC_ISI_PM_SUSPENDED)
+	if ((mxc_isi->flags & MXC_ISI_PM_SUSPENDED) ||
+		(mxc_isi->flags & MXC_ISI_RUNTIME_SUSPEND))
 		return 0;
 
 	clk_disable_unprepare(mxc_isi->clk);
-- 
1.7.9.5

